### PR TITLE
chore: add Server-side README

### PR DESCRIPTION
**Pull request by cwaldren-ld**
Adds a new README for the server. 

Also adds a link from the client-side README to this one.

### CODE REVIEW

File: libs/server-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+LaunchDarkly Server-Side SDK for C/C++
+===================================
+
+[![Actions Status](https://github.com/launchdarkly/cpp-sdks/actions/workflows/server.yml/badge.svg)](https://github.com/launchdarkly/cpp-sdks/actions/workflows/server.yml)
+[![Documentation](https://img.shields.io/static/v1?label=GitHub+Pages&message=API+reference&color=00add8)](https://launchdarkly.github.io/cpp-sdks/libs/server-sdk/docs/html/)
+
+The LaunchDarkly Server-Side SDK for C/C++ is designed primarily for use in in multi-user systems such as web servers
```
Comment (mmrj) [inline libs/server-sdk/README.md:7]:
```suggestion
The LaunchDarkly Server-Side SDK for C/C++ is designed primarily for use in multi-user systems such as web servers
```
some day this kind of thing will be solved by `sdk-docs` 🤞

File: libs/server-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+and applications. It follows the server-side LaunchDarkly model for multi-user contexts.
+It is not intended for use in desktop and embedded systems applications.
+
+For using LaunchDarkly in client-side C/C++ applications, refer to our [Client-Side C/C++ SDK](../client-sdk/README.md).
+
+LaunchDarkly overview
+-------------------------
+[LaunchDarkly](https://www.launchdarkly.com) is a feature management platform that serves over 100 billion feature flags
```
Comment (mmrj) [inline libs/server-sdk/README.md:15]:
```suggestion
[LaunchDarkly](https://www.launchdarkly.com) is a feature management platform that serves trillions of feature flags
```
[latest](https://github.com/launchdarkly/sdk-docs/blob/main/docs/readme-sdk/v1/items/readme-preamble.txt#L17)

File: libs/server-sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+the [Github releases](https://github.com/launchdarkly/cpp-sdks/releases?q=cpp-server&expanded=true) for use in your
+project.
+
+Refer to the [SDK documentation][reference-guide] for complete instructions on
+installing and using the SDK.
+
+### Incorporating the SDK
+
+The SDK can be used via a C++ or C interface and can be incorporated via a static library or shared object. The static
+library and shared object each have their on use cases and limitations.
```
Comment (mmrj) [inline libs/server-sdk/README.md:39]:
```suggestion
library and shared object each have their own use cases and limitations.
```

File: libs/server-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+We run integration tests for all our SDKs using a centralized test harness. This approach gives us the ability to test
+for consistency across SDKs, as well as test networking behavior in a long-running application. These tests cover each
+method in the SDK, and verify that event sending, flag evaluation, stream reconnection, and other aspects of the SDK all
+behave correctly.
+
+Contributing
+------------
+
+We encourage pull requests and other contributions from the community. Check out
```
Comment (mmrj) [inline libs/server-sdk/README.md:92]:
```suggestion
We encourage pull requests and other contributions from the community. Read
```
[latest](https://github.com/launchdarkly/sdk-docs/blob/main/docs/readme-sdk/v1/items/readme-trailer.txt#L9C72-L9C76)

File: libs/server-sdk/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+our [contributing guidelines](../../CONTRIBUTING.md) for instructions on how to contribute to this SDK.
+
+About LaunchDarkly
```
Comment (mmrj) [inline libs/server-sdk/README.md:95]: (This section also has some minor changes in the `sdk-docs` repo, but we haven't rolled them out anywhere, so I think it's fine to leave this as-is.)

**Review by mmrj — CHANGES_REQUESTED** [review]

File: libs/server-sdk/README.md (hunk 2 above)
Comment (cwaldren-ld) [reply libs/server-sdk/README.md:15]: 😅 

More motivation day by day..

**Review by mmrj — APPROVED** [review]

**Pull request merged**